fatoptimizer/methods.py

```diff
@@ -35,6 +41,44 @@ def add(obj_type, name, *args, **kw):
     add(bytes, 'decode', (0, 2), str, str,
         check_args=check_encoding,
         exceptions=UnicodeDecodeError)
+    add(bytes, 'endswith', (1, 3), tuple, int, int)
+    add(bytes, 'find', (1, 3), bytes, int, int)     # FIXME: Here "sub"(that is to be found) can be both bytes and int
+    add(bytes, 'index', (1, 3), bytes, int, int)    # FIXME: Here "sub" can be both bytes and int
+
+    #FIXME: How to specify any iterable?
```

Comment:
I'm not sure that it's safe to optimize set and frozenset types, since these containers are unordered, and so it's probably better to optimize them at runtime.

Maybe just use the "object" type and use your check_bytetype checker?